umake/frameworks/swift.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2014 Canonical
+#
+# Authors:
+#  Didier Roche
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the Free Software
+# Foundation; version 3.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+
+
+"""Swift module"""
+
+from contextlib import suppress
+from gettext import gettext as _
+import logging
+import os
+import re
+import gnupg
+
+import umake.frameworks.baseinstaller
+from umake.interactions import DisplayMessage
+from umake.tools import add_env_to_user, MainLoop, get_current_ubuntu_version
+from umake.network.download_center import DownloadCenter, DownloadItem
+from umake.ui import UI
+
+logger = logging.getLogger(__name__)
+
+
+class SwiftCategory(umake.frameworks.BaseCategory):
+
+    def __init__(self):
+        super().__init__(name="Swift", description=_("Swift language"),
+                         logo_path=None)
+
+
+class SwiftLang(umake.frameworks.baseinstaller.BaseInstaller):
+
+    def __init__(self, category):
+        super().__init__(name="Swift Lang", description=_("Swift compiler (default)"), is_category_default=True,
+                         packages_requirements=["clang"],
```

Comment:
From https://swift.org/download, I guess you need to add libicu-dev as well (even if installed by default, in case the user removed it). I'm going to add it to the docker file as well.